### PR TITLE
Animate StatusBar background color

### PR DESCRIPTION
**Pull request by roryabraham**
### Details
This PR stemmed off of https://github.com/Expensify/App/pull/15462. For now, the only visual change is that the status bar on the homepage w/ the sidebar open matches the sidebar background color

#### Before
https://github.com/Expensify/App/assets/47436092/405e5c36-35f6-4066-880e-03da3da2f6da

#### After
https://github.com/Expensify/App/assets/47436092/6f9f65a7-5348-448d-826b-aa8068e1ff86

It's implemented in a very generic way, where the same effect can take place on any screen just by adding to the `PAGE_BACKGROUND_COLOR` constant in the theme file, because we'll be using it more in the future.

Note that this really only has any effect on mobile web, and not all browsers are even affected:

- iOS (native) app displays edge-to-edge, so the status bar is transparent
- Android (native) as of very recently, the app displays edge-to-edge, to the status bar is transparent
- macOS Safari – unaffected by `StatusBar.setBackgroundColor`
- macOS Chrome – unaffected
- desktop app – unaffected
- iOS Chrome – unaffected
- iOS Safari – this is the primary place where this change can be observed.
- Android Chrome – my Pixel 4a w/ Chrome is unaffected, but it's possible other devices / browsers may be affected.

### Fixed Issues
$ (partial) https://github.com/Expensify/App/issues/12261

### Tests / QA Steps (iOS Safari)
1. Run the app
1. Verify that the while you're looking at the chat list / sidebar, the status bar matches the background
1. Open a regular chat. Verify that the StatusBar animates to match the new page background (note: IOU reports happen to have a different background near the top of the screen than a regular chat, I'm considering this out-of-scope for now).
1. Go back to the sidebar. Verify that the StatusBar animates to match the sidebar background.
1. Open the settings page. Verify that the StatusBar animates to match the settings page background.
1. Go back to the sidebar. Verify that the StatusBar animates to match the sidebar background.

### Tests / QA Steps (all other platforms)
Nothing – just regression tests. Let me know if you see different behavior in an Android browser. Most likely it will not be a blocker, but since there are a lot of different Android devices I suppose it's possible something unexpected could happen 🤷🏼

- [x] Verify that no errors appear in the JS console

### Offline tests
Repeat all the same steps offline – it should all happen the same.

### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

no effect

</details>

<details>
<summary>Mobile Web - Chrome</summary>

no effect

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/47436092/6f9f65a7-5348-448d-826b-aa8068e1ff86

</details>

<details>
<summary>Desktop</summary>

no effect

</details>

<details>
<summary>iOS</summary>

no effect

</details>

<details>
<summary>Android</summary>

no effect

</details>

### CODE REVIEW

**Comment by roryabraham** [comment]
@sobitneupane requesting your review since you were assigned as C+ in the [original PR](https://github.com/Expensify/App/pull/15462) and [issue](https://github.com/Expensify/App/issues/12261)

**Review by sobitneupane — COMMENTED** [review]
Getting few console errors in Web:
Screenshot 2023-06-14 at 12 23 57
Screenshot 2023-06-14 at 12 23 49
Screenshot 2023-06-14 at 12 23 27
Screenshot 2023-06-14 at 12 23 16
Screenshot 2023-06-14 at 12 23 39

Screenshot 2023-06-14 at 12 24 04

**Review by sobitneupane — COMMENTED** [review]
In addition to Money Request Header, header in Task Page and AttachmentModal has lighter background color and doesn't match with statusBar. I think they are also out of scope of this PR as mentioned in the Test steps.

**Comment by roryabraham** [comment]
@sobitneupane Unless I'm mistaken, all those propTypes errors were not introduced by this code.

**Comment by roryabraham** [comment]
> In addition to Money Request Header, header in Task Page and AttachmentModal has lighter background color and doesn't match with statusBar. I think they are also out of scope of this PR as mentioned in the Test steps.

Agree those are out of scope. As stated in the PR description, this PR is implemented in a way that should make it very easy to fix similar issues in the future by just adding the screen + background color to the style file.

**Comment by roryabraham** [comment]
Updated and conflicts resolved.

File: src/components/withCurrentReportId.js [hunk 1]
```diff
@@ -9,6 +9,10 @@ const CurrentReportIdContext = createContext(null);
 const withCurrentReportIdPropTypes = {
     /** Actual content wrapped by this component */
     children: PropTypes.node.isRequired,
+
+    /** Function used to update the current reportId. Should probably only ever be used in one place in the whole app. */
+    // eslint-disable-next-line react/no-unused-prop-types
+    updateCurrentReportId: PropTypes.func.isRequired,
```
Comment (sobitneupane) [inline src/components/withCurrentReportId.js:15]: @roryabraham I think those console errors are related to this change.

**Comment by roryabraham** [comment]
@sobitneupane the console errors were only related in that I was using a HOC that didn't have its propTypes correctly defined. Anyways, I've since dropped the HOC and just used a hook instead, so it shouldn't be a problem anymore. Going to retest and confirm.

**Comment by roryabraham** [comment]
Verified that the console errors are fixed ✅

**Review by sobitneupane — COMMENTED** [review]
It solves the issue both on IOS/safari and Android/Chrome

### Screenshots/Videos
<details>
<summary>Web</summary>
Screenshot 2023-06-23 at 12 24 44


</details>

<details>
<summary>Mobile Web - Chrome</summary>


https://github.com/Expensify/App/assets/25876548/894b72c5-6170-4335-91a1-028443d2d70e


</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/25876548/77134f9c-ff54-4b56-b14f-8e57cdd4355c



</details>

<details>
<summary>Desktop</summary>
Screenshot 2023-06-23 at 12 34 16


</details>

<details>
<summary>iOS</summary>


Screenshot 2023-06-23 at 12 29 52

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/25876548/95298195-b7c4-47fe-bed3-afd06b6a4c32


</details>

**Review by sobitneupane — COMMENTED** [review]
@roryabraham I am getting following console error on IOS and android. I am not sure if it is related to this PR or not but it is not reproducible in main. If it is not reproducilbe in your end we are good to merge this.

Screenshot 2023-06-23 at 12 31 34

File: src/libs/Navigation/NavigationRoot.js [hunk 2]
```diff
@@ -56,18 +54,54 @@ function NavigationRoot(props) {
...
+
+    const prevStatusBarBackgroundColor = useRef(themeColors.appBG);
+    const statusBarBackgroundColor = useRef(themeColors.appBG);
+    const statusBarAnimation = useSharedValue(0);
+
+    useAnimatedReaction(
+        () => statusBarAnimation.value,
+        () => {
+            const color = interpolateColor(statusBarAnimation.value, [0, 1], [prevStatusBarBackgroundColor.current, statusBarBackgroundColor.current]);
+            runOnJS(StatusBar.setBackgroundColor)(color);
```
Comment (sobitneupane) [inline src/libs/Navigation/NavigationRoot.js:68]: Looks like above console error is related to `runOnJS`.

**Comment by roryabraham** [comment]
@sobitneupane fixed the console error 👍🏼

**Comment by roryabraham** [comment]
@sobitneupane that one is safe to ignore since our app displays edge-to-edge.

**Comment by roryabraham** [comment]
@sobitneupane resolved conflicts and that warning

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>
Screenshot 2023-06-29 at 14 12 16

</details>

<details>
<summary>Mobile Web - Chrome</summary>



https://github.com/Expensify/App/assets/25876548/89695a80-09d7-4a3e-b7d8-079b4e28d331



</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/25876548/1839db1f-f257-4301-a940-ed3ddccd774a


</details>

<details>
<summary>Desktop</summary>
Screenshot 2023-06-29 at 14 26 57



</details>

<details>
<summary>iOS</summary>



https://github.com/Expensify/App/assets/25876548/783e42b5-75ce-4b48-9072-c0d0fa0bb081


</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/25876548/3ea1a3ca-3ca5-438c-9f29-96ac8707d399




</details>

**Review by AndrewGable — APPROVED** [review]

**Comment by sobitneupane** [comment]
@roryabraham @AndrewGable PR was deployed to production 3 weeks ago. Should I make payment request on newDot?

**Comment by AndrewGable** [comment]
Yes

**Comment by sobitneupane** [comment]
Thanks @AndrewGable. Requested payment on newDot.

**Comment by JmillsExpensify** [comment]
@AndrewGable Can you confirm that the correct payment for @sobitneupane is $1,000 for this PR review?

**Comment by AndrewGable** [comment]
Yes 👍

**Comment by JmillsExpensify** [comment]
Reviewed details for @sobitneupane. These details are accurate based on summary from Business Reviewer (note: No BugZero member assigned to this issue) and are now approved for payment in NewDot.

**Pull request merged**